libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -240,6 +257,8 @@ void WasmDialect::addFunction(
 	yulAssert(_returns.size() <= 1, "The Wasm 1.0 specification only allows up to 1 return value.");
 	f.returns = std::move(_returns);
 	f.sideEffects = _movable ? SideEffects{} : SideEffects::worst();
+	f.sideEffects.cannotLoop = true;
+	f.sideEffects.movableApartFromEffects = true;
```

Comment:
Why are these 2 above true?

Reply:
`cannotLoop` is true for all builtins. This property is set in `SideEffectCollector`if there is a loop/recursion in function control flow.

I think I'll set `movableApartFromEffects` to `_movable` and add a TODO comment that this needs to be fixed when YUL-optimizations for WASM get properly done.